### ISSUE TITLE
NullPointerException in io.swagger.jackson.ModelResolver#_addEnumProps in swagger-core 1.6.2-SNAPSHOT

### ISSUE THREAD

**Issue by blop** [swagger-api/swagger-core#3605]
Hello,

I have the following error :
```
java.lang.NullPointerException: null
	at io.swagger.jackson.ModelResolver._addEnumProps(ModelResolver.java:212)
	at io.swagger.jackson.ModelResolver.resolveProperty(ModelResolver.java:170)
	at io.swagger.jackson.ModelResolver.resolveProperty(ModelResolver.java:126)
	at io.swagger.converter.ModelConverterContextImpl.resolveProperty(ModelConverterContextImpl.java:83)
```

The code failing is : 
```
        Class<Enum<?>> enumClass = (Class<Enum<?>>) propClass;
        for (Enum<?> en : enumClass.getEnumConstants()) {
```

java.lang.Class#getEnumConstants states that it returns "an array containing the values comprising the enum class represented by this Class object in the order they're declared, **or null if this Class object does not represent an enum type**"

So I guess it should be null checked before iterating on it.
It fails on a field of type Enum<?> in our code.

It was working in 1.5.17 (did not try other versions).

**Comment by frantuma** [comment]
Thanks for reporting this and for the PR. fixed in #3607

**Comment by giylmi** [comment]
Hey, I got this issue in `1.6.2`. I see that the PR hadn't been merged to `1.6.2`. Is there a plan to include this PR in next release?

**Comment by frandelgado** [comment]
Also seing this issue in `1.6.2`

**Comment by nikita-incorta** [comment]
hello, Im seeing this issue in 1.6.2, is there any plan to fix it in 1.6.2?

**Comment by Splash34** [comment]
+1

**Comment by Splash34** [comment]
Fixed with version 1.6.6
